client/highlevel-rest/src/main/java/org/elasticsearch/client/ClientSearchHit.java

```diff
@@ -185,80 +189,97 @@ public Explanation explanation() {
 
     @Override
     public Explanation getExplanation() {
-        // TODO
-        return null;
+        return explanation();
     }
 
     @Override
     public SearchHitField field(String fieldName) {
-        // TODO
-        return null;
+        return fields().get(fieldName);
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public Map<String, SearchHitField> fields() {
-        // TODO
-        return null;
+        Map<String, Object> originalMap = (Map<String, Object>) this.objectPath.evaluate("fields");
+        Map<String, SearchHitField> fields = new HashMap<>(originalMap.size());
+        for (Entry<String, Object> original : originalMap.entrySet()) {
+            fields.put(original.getKey(), new InternalSearchHitField(original.getKey(), (List<Object>) original.getValue()));
+        }
+        return fields;
     }
 
     @Override
     public Map<String, SearchHitField> getFields() {
-        // TODO
-        return null;
+        return fields();
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public Map<String, HighlightField> highlightFields() {
-        // TODO
-        return null;
+        Map<String, Object> originalMap = (Map<String, Object>) this.objectPath.evaluate("highlight");
+        Map<String, HighlightField> fields = new HashMap<>(originalMap.size());
+        for (Entry<String, Object> original : originalMap.entrySet()) {
+            List<String> fragments = (List<String>) original.getValue();
+            Text[] asText = new Text[fragments.size()];
+            int i = 0;
+            for (String fragment : fragments) {
+                asText[i] = new Text(fragment);
+                i++;
+            }
+            fields.put(original.getKey(), new HighlightField(original.getKey(), asText));
+        }
+        return fields;
     }
 
     @Override
     public Map<String, HighlightField> getHighlightFields() {
-        // TODO
-        return null;
+        return highlightFields();
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public Object[] sortValues() {
-        // TODO
-        return null;
+        return ((List<Object>) this.objectPath.evaluate("sort")).toArray();
     }
 
     @Override
     public Object[] getSortValues() {
-        // TODO
-        return null;
+        return sortValues();
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public String[] matchedQueries() {
-        // TODO
-        return null;
+        List<String> matched = (List<String>) this.objectPath.evaluate("matched_queries");
+        return matched.toArray(new String[matched.size()]);
     }
 
     @Override
     public String[] getMatchedQueries() {
-        // TODO
-        return null;
+        return matchedQueries();
     }
 
     @Override
     public SearchShardTarget shard() {
-        // TODO
+        // TODO, not sure this will work
```

Comment:
Okay, should I remove this (and implementing the interface) in this PR or just change the TODO?